Sources/XMTP/PreparedMessage.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  PreparedMessage.swift
+//
+//
+//  Created by Pat Nakajima on 3/9/23.
+//
+
+import CryptoKit
+import Foundation
+
+public struct PreparedMessage {
+	var messageEnvelope: Envelope
+	var conversation: Conversation
+	var onSend: () async throws -> Void
+
+	public func decodedMessage() throws -> DecodedMessage {
+		return try conversation.decode(messageEnvelope)
```

Comment:
Could maybe have prepared message include EncodedContent and stuff so we don't need to involve the encryption stuff here, but this seems like an OK start for now.